test/libgit2.jl

```diff
@@ -199,12 +199,20 @@ mktempdir() do dir
                 @test isa(remote, LibGit2.GitRemote)
                 @test sprint(show, remote) == "GitRemote:\nRemote name: upstream url: $repo_url"
                 @test LibGit2.isattached(repo)
+                LibGit2.set_remote_url(repo, "", remote="upstream")
+                remote = LibGit2.get(LibGit2.GitRemote, repo, branch)
+                @test sprint(show, remote) == "GitRemote:\nRemote name: upstream url: "
+                close(remote)
+                LibGit2.set_remote_url(cache_repo, repo_url, remote="upstream")
+                remote = LibGit2.get(LibGit2.GitRemote, repo, branch)
+                @test sprint(show, remote) == "GitRemote:\nRemote name: upstream url: $repo_url"
                 close(remote)
 
                 remote = LibGit2.GitRemoteAnon(repo, repo_url)
                 @test LibGit2.url(remote) == repo_url
                 @test LibGit2.name(remote) == ""
                 @test isa(remote, LibGit2.GitRemote)
+                close(remote)
```

Comment:
The `finally` closes the `repo`, which owns this remote, right? But what if someone adds tests below this and forgets to close the anon-remote but still wants to modify the remote with the same url? That was my thinking.